app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -141,6 +145,10 @@ public AppLanguageLookUpTable getLanguageLookUpTable() {
      */
     public static Map<String, Boolean> pauseUploads = new HashMap<>();
 
+    /**
+     *  In-memory list of contributions whose uploads have not been properly fnished
+     */
+    public static Map<String, Boolean> unfinishedUploads = new HashMap<>();
```

Comment:
Better avoid public members, especially for modifyable data structures. Better make it private and add public methods to modify the data in intended ways.